File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/infusion/InfusionRepository.java
package coolpharaoh.tee.speicher.tea.timer.core.infusion;

import android.app.Application;

import java.util.List;

import coolpharaoh.tee.speicher.tea.timer.database.TeaMemoryDatabase;

public class InfusionRepository {

    private final InfusionDao infusionDao;

    public InfusionRepository(final Application application) {
        final TeaMemoryDatabase database = TeaMemoryDatabase.getDatabaseInstance(application);
        infusionDao = database.getInfusionDao();
    }

    public void insertInfusion(final Infusion infusion) {
        infusionDao.insert(infusion);
    }

    public List<Infusion> getInfusions() {
        return infusionDao.getInfusions();
    }

    public List<Infusion> getInfusionsByTeaId(final long id) {
        return infusionDao.getInfusionsByTeaId(id);
    }

    public void deleteInfusionsByTeaId(final long id) {
        infusionDao.deleteInfusionsByTeaId(id);
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/infusion/InfusionRepository.kt
package coolpharaoh.tee.speicher.tea.timer.core.infusion

import android.app.Application
import coolpharaoh.tee.speicher.tea.timer.database.TeaMemoryDatabase

class InfusionRepository(application: Application?) {
    private val infusionDao: InfusionDao

    init {
        val database = TeaMemoryDatabase.getDatabaseInstance(application)
        infusionDao = database.infusionDao
    }

    fun insertInfusion(infusion: Infusion) {
        infusionDao.insert(infusion)
    }

    val infusions: List<Infusion>
        get() = infusionDao.getInfusions()

    fun getInfusionsByTeaId(id: Long): List<Infusion> {
        return infusionDao.getInfusionsByTeaId(id)
    }

    fun deleteInfusionsByTeaId(id: Long) {
        infusionDao.deleteInfusionsByTeaId(id)
    }
}